packages/AdaptiveCards/Actions/SendDataQueryResponse.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using AdaptiveCards.Templating;
+using AdaptiveExpressions.Properties;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Components.AdaptiveCards
+{
+    public class SendDataQueryResponse : Dialog
+    {
+        [JsonConstructor]
+        public SendDataQueryResponse([CallerFilePath] string sourceFilePath = "", [CallerLineNumber] int sourceLineNumber = 0)
+            : base()
+        {
+            // enable instances of this command as debug break point
+            this.RegisterSourceLocation(sourceFilePath, sourceLineNumber);
+        }
+
+        [JsonProperty("$kind")]
+        public const string Kind = "Microsoft.Bot.Components.SendDataQueryResponse";
+
+        [JsonProperty("disabled")]
+        public BoolExpression Disabled { get; set; }
+
+        [JsonProperty("results")]
+        public ArrayExpression<object> Results { get; set; }
+
+        public async override Task<DialogTurnResult> BeginDialogAsync(DialogContext dc, object options = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (options is CancellationToken)
```

Comment:
I'm seeing this logic repeated a lot. Is there any way we could centralize and reutilize instead of copy+pasting?

Reply:
I took this from the SDK... It's all over the place there as well.

---
In reply to: [610224728](https://github.com/microsoft/botframework-components/pull/862#discussion_r610224728) [](ancestors = 610224728)